### DISCUSSION TITLE
v23.0.0-beta.1

### DISCUSSION THREAD

**Discussion by thaJeztah** [moby/moby#44589] [Announcements]
This is a pre-release of the upcoming 23.0.0 release. 

Pre-releases are intended for testing new releases: **only install in a test environment!**

```bash
curl -fsSL https://get.docker.com -o get-docker.sh
sudo CHANNEL=test sh get-docker.sh
```

Starting with the 23.0.0 release, we're moving away from using CalVer versioning,
and use versioning that's using SemVer _format_. Changing the version format is
a stepping-stone towards go module compatibility, but the repository does not
yet use go-modules, and still requires using a "+incompatible" version; we'll
be working towards go module compatibility in a future release.

### Known issues:

- There is no changelog yet; an overview of pull requests included in this release can be found on GitHub:
    - docker cli: [all pull requests for 23.0.0](https://github.com/docker/cli/pulls?q=is%3Apr+milestone%3A23.0.0+is%3Amerged) / [all "changelog" pull requests for 23.0.0](https://github.com/docker/cli/pulls?q=is%3Apr+milestone%3A23.0.0+label%3Aimpact%2Fchangelog+is%3Amerged)
    - docker engine: [all pull requests for 23.0.0](https://github.com/moby/moby/pulls?q=is%3Apr+milestone%3A23.0.0+is%3Amerged) / [all "changelog" pull requests for 23.0.0](https://github.com/moby/moby/pulls?q=is%3Apr+milestone%3A23.0.0+label%3Aimpact%2Fchangelog+is%3Amerged)
- Static binaries (`.tar.gz`) are no longer include cli plugins; use the deb/rpm packages, or manual installation instructions from the [buildx github repository](https://github.com/docker/buildx#manual-download) and  [docker compose github repository](https://github.com/docker/compose/#where-to-get-docker-compose)
- The `docker-proxy` binary is not statically linked in the static packages
- There are no packages available yet for the s390x and ppc64le architectures
- There may be bugs! :warning:

Bugs and regressions can be reported in these issue trackers:

- Related to the CLI: https://github.com/docker/cli/issues
- Related to the Docker Engine https://github.com/moby/moby/issues

When reporting issues, include `[23.0.0-beta]` in the issue title

<hr /><em>This discussion was created from the release <a href='https://github.com/moby/moby/releases/tag/v23.0.0-beta.1'>v23.0.0-beta.1</a>.</em>

**Comment by firecow** [comment]
Will pull requests in https://github.com/moby/swarmkit be included?

**Reply by thaJeztah** [reply]
The version of the SwarmKit dependency used for the release can be found in the `vendor.mod` file; for v23.0.0-beta.1; this is the version; https://github.com/moby/moby/blob/v23.0.0-beta.1/vendor.mod#L57

Which corresponds with this commit: https://github.com/moby/swarmkit/commit/b17f02f0a0547a2416ce3bf2f59d9050288ec31b

**Comment by slonopotamus** [comment]
Wait, but are we going to see 22.06.x releases? [This](https://github.com/moby/moby/releases/tag/v22.06.0-beta.0) said it is "the first pre-release of the upcoming 22.06 release".

**Reply by thaJeztah** [reply]
23.0.0 is what's "formerly known as "22.06";

> Starting with the 23.0.0 release, we're moving away from using CalVer versioning,
and use versioning that's using SemVer format. Changing the version format is
a stepping-stone towards go module compatibility, but the repository does not
yet use go-modules, and still requires using a "+incompatible" version; we'll
be working towards go module compatibility in a future release.

**Comment by s4ke** [comment]
This is exciting :). Congrats on the (beta) release!

**Comment by meox** [comment]
This is on a Debian 11 Testing

```
# Executing docker install script, commit: 4f282167c425347a931ccfd95cc91fab041d414f
Warning: the "docker" command appears to already exist on this system.

If you already have Docker installed, this script can cause trouble, which is
why we're displaying this warning and provide the opportunity to cancel the
installation.

If you installed the current Docker package using this script and are using it
again to update Docker, you can safely ignore this message.

You may press Ctrl+C now to abort this script.
+ sleep 20
+ sh -c apt-get update -qq >/dev/null
+ sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y -qq apt-transport-https ca-certificates curl >/dev/null
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = "en_US:en",
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LC_TERMINAL = "iTerm2",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
+ sh -c mkdir -p /etc/apt/keyrings && chmod -R 0755 /etc/apt/keyrings
+ sh -c curl -fsSL "https://download.docker.com/linux/debian/gpg" | gpg --dearmor --yes -o /etc/apt/keyrings/docker.gpg
+ sh -c chmod a+r /etc/apt/keyrings/docker.gpg
+ sh -c echo "deb [arch=arm64 signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian bookworm test" > /etc/apt/sources.list.d/docker.list
+ sh -c apt-get update -qq >/dev/null
E: The repository 'https://download.docker.com/linux/debian bookworm Release' does not have a Release file.
```

**Reply by thaJeztah** [reply]
That's expected currently; Debian "testing" is not Debian 11, but Debian 12 ("bookworm" is the codename for Debian 12); https://wiki.debian.org/DebianReleases

We don't have packages yet for that release, as it's still under development, and the convenience install script does not know about that code name yet.

You may be able to try installing the packages for debian 11 ("bullseye"); either manually, or by downloading the script and patching it to map "bookworm" to "bullseye". Definitely not tested though, so no promises, but should be fine for testing (and happy to hear if there issues you run into)

**Comment by crazy-max** [comment]
> * The `docker-proxy` binary is not statically linked in the static packages

Looks good now with https://github.com/moby/moby/pull/44546

See https://github.com/moby/moby/actions/runs/3711183189/jobs/6291363410#step:7:9

> There are no packages available yet for the s390x and ppc64le architectures

If some of you are willing to test the [experimental packaging](https://github.com/docker/packaging) provided through a docker image (s390x included but not ppc64le yet):
* engine: https://github.com/docker/packaging/releases/tag/docker-engine%2Fv23.0.0-beta.1-14
* cli: https://github.com/docker/packaging/releases/tag/docker-cli%2Fv23.0.0-beta.1-14
